### PR TITLE
DT and RF user doc updates

### PR DESCRIPTION
**Pull request by fmcquillan99**
@rahiyer please review DT user doc updates

Will start working on RF in parallel.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/385/

File: src/ports/postgres/modules/recursive_partitioning/decision_tree.sql_in [hunk 1]
```diff
@@ -418,7 +468,10 @@ tree_predict(tree_model,
   <DT>new_data_table</DT>
   <DD>TEXT. Name of the table containing prediction data. This table is
   expected to contain the same features that were used during training. The table
-  should also contain <em>id_col_name</em> used for identifying each row.</DD>
+  should also contain <em>id_col_name</em> used for identifying each row.
+
+  If the <em>new_data_table</em> contains categorical variables
```
[the remark this answers is not in the document]
Comment (fmcquillan99) [reply src/ports/postgres/modules/recursive_partitioning/decision_tree.sql_in]: Ok, I will remove this line.

**Comment by fmcquillan99** [comment]
Waiting on these bugs to be fixed before I can finish:
https://issues.apache.org/jira/browse/MADLIB-1217
https://issues.apache.org/jira/browse/MADLIB-1218
https://issues.apache.org/jira/browse/MADLIB-1219

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/397/

**Comment by fmcquillan99** [comment]
@rahiyer RF docs ready for review too.

File: src/ports/postgres/modules/recursive_partitioning/decision_tree.sql_in [hunk 2]
```diff
@@ -127,18 +128,20 @@ tree_train(
...
+  each observation.  Can be any value greater 
+  than 0 (does not need to be an integer).  
   This can be used to handle the case of unbalanced data sets.
-  For classification the row's vote is multiplied by the weight, 
-  and for regression we perform a weighted average at each node.
-  If this parameter is not set, all observations (tuples)
-  are treated equally with a weight of 1.0.</DD>
+  The weights are used to compute a weighted average in 
+  the output leaf node. For classification, the contribution 
+  of a row towards the vote of it's corresponding level
```
Comment (iyerr3) [inline src/ports/postgres/modules/recursive_partitioning/decision_tree.sql_in]: `s/it's/its`. I had a typo in my comment as well.

File: src/ports/postgres/modules/recursive_partitioning/random_forest.sql_in [hunk 3]
```diff
@@ -97,327 +264,220 @@ forest_train(training_table_name,
...
     <tr>
     <th>num_failed_groups</th>
-    <td>int. Number of failed groups during forest training.</td>
+    <td>INTEGER. Number of failed groups during forest training.</td>
     </tr>
 
     <tr>
       <th>total_rows_processed</th>
-      <td>bigint. Total numbers of rows processed in all groups.</td>
+      <td>BIG INTEGER. Total numbers of rows processed in all groups.</td>
```
Comment (iyerr3) [inline src/ports/postgres/modules/recursive_partitioning/random_forest.sql_in]: This is `BIGINT`. Postgres doesn't expand on the `INT` for this type. Same comment for the next item as well.

**Comment by fmcquillan99** [comment]
https://issues.apache.org/jira/browse/MADLIB-1217
https://issues.apache.org/jira/browse/MADLIB-1218
https://issues.apache.org/jira/browse/MADLIB-1219
have all been fixed so I made associated changes to the DT and RF user docs.

I think this is final, I mean, I don't have any other commits to make as part of this PR.

So after final review please merge if you think it looks OK.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/405/

**Pull request closed without merging**